### PR TITLE
#149-Split_AWS_tests_part_2

### LINKED ISSUE #149: Split AWS Tests
- the validate CloudFormation template tests need AWS credentials; hence they are integration tests
- the lint CloudFormation template tests can run locally, but are a bit slower, but could still be part of the unit tests
  - calling the local linter takes a few seconds
- the aws tests that use mocks are likely unit tests and should move there  

Delete file `test/codebuild/test_install_dependencies.py`

E.g. in file `test/integration/aws/test_deploy_vm_bucket.py`:
- `test_deploy_vm_bucket_template`
  - needs credentials -> `integration`
- `test_deploy_vm_bucket_template_with_cnf_lint`
  - does not need aws credentials -> `unit`

### LINKED ISSUE #200: Replace version "latest" of localstack by concrete version number
During implementation of #187 showed problems with latest concrete versions of localstack on docker hub, e.g. 3.1.0.

Explicit version `latest` in contrast fixes the bug but for reliable robust tests, we would prefer to use a concrete version of [localstack](https://hub.docker.com/r/localstack/localstack/tags) on docker hub.

As soon as such a version `<V>`is available, the current ticket requests to replace version `latest` in the [ai-lab tests](https://github.com/exasol/ai-lab/blob/main/test/integration/localstack_test.py) by `<V>.`

### PR DESCRIPTION
**Pull request by ckunki**
checkout files from other branch

closes #200 
closes #149

### CODE REVIEW

File: test/unit/aws/conftest.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from test.aws.conftest import (
```
Comment (tkilias) [inline test/unit/aws/conftest.py:1]: this is a bad idea, don't import from conftest, if you want to externalize fixtures use https://docs.pytest.org/en/7.1.x/how-to/writing_plugins.html#requiring-loading-plugins-in-a-test-module-or-conftest-file

File: test/unit/aws/test_deploy_ec2.py [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
-    aws_access.validate_cloudformation_template(ec2_cloudformation_yml)
-
-
-def test_deploy_cec2_template_with_cnf_lint(tmp_path, ec2_cloudformation_yml):
-    validate_using_cfn_lint(tmp_path, ec2_cloudformation_yml)
+    aws = MagicMock()
+    aws.stack_exists.return_value = False
+    with CloudformationStackContextManager(
+            CloudformationStack(
+                aws, "test_key", "test_user", default_asset_id, test_dummy_ami_id,
```
Comment (tkilias) [inline test/unit/aws/test_deploy_ec2.py:41]: above each parameter was in a seperate line

File: test/unit/aws/test_deploy_vm_bucket.py [hunk 3]
```diff
@@ -4,63 +4,56 @@
...
+    mock_cast(aws.instantiate_for_region).return_value = aws
+    run_setup_vm_bucket(aws, test_config)
+    mock_cast(aws.upload_cloudformation_stack).assert_called_once()
+
+    bucket = find_vm_bucket(aws)
     assert TEST_BUCKET_ID == bucket
 
 
-def test_find_fails_if_vm_stack_not_deployed_with_mock(test_config):
+def test_vm_bucket_undeployed(test_config):
```
Comment (tkilias) [inline test/unit/aws/test_deploy_vm_bucket.py:37]: undeployed is a bit unusual

**Review by tkilias — CHANGES_REQUESTED** [review]

File: test/unit/aws/conftest.py (hunk 1 above)
Comment (ckunki) [reply test/unit/aws/conftest.py:1]: I will replace the import by plugins in next push.

File: test/unit/aws/test_deploy_ec2.py (hunk 2 above)
Comment (ckunki) [reply test/unit/aws/test_deploy_ec2.py:41]: I did put all parameters in a single line here, as it felt short enough.

File: test/integration/aws/conftest.py [hunk 4]
```diff
@@ -5,11 +5,8 @@
 
 from importlib.metadata import version
 from test.aws.local_stack_access import AwsLocalStackAccess
-from test.aws.conftest import (
-    default_asset_id,
-    test_dummy_ami_id,
-    ec2_cloudformation_yml,
-)
+
+pytest_plugins = "test.aws.fixtures"
```
Comment (tkilias) [inline test/integration/aws/conftest.py:9]: pytest_plugins is deprecated for not-root conftest
https://docs.pytest.org/en/7.1.x/how-to/writing_plugins.html#requiring-loading-plugins-in-a-test-module-or-conftest-file

File: test/unit/aws/conftest.py [hunk 5]
```diff
@@ -1,5 +1 @@
-from test.aws.conftest import (
-    default_asset_id,
-    ec2_cloudformation_yml,
-    test_dummy_ami_id,
-)
+pytest_plugins = "test.aws.fixtures"
```
Comment (tkilias) [inline test/unit/aws/conftest.py:1]: pytest_plugins is deprecated for not-root conftest
https://docs.pytest.org/en/7.1.x/how-to/writing_plugins.html#requiring-loading-plugins-in-a-test-module-or-conftest-file

**Review by tkilias — CHANGES_REQUESTED** [review]

File: test/unit/aws/conftest.py (hunk 5 above)
Comment (ckunki) [reply test/unit/aws/conftest.py]: yes, I read it first and decided to ignore it, but the failed ci build seems to enforce the deprecation. 
Please see latest push.

**Pull request merged**